quickwit-search/src/search_stream/mod.rs

```diff
@@ -68,6 +91,37 @@ fn serialize_click_house_row_binary<TFastValue: FastValue + Display>(
     Ok(())
 }
 
+mod helpers {
+    use std::fmt::Display;
+
+    use tantivy::fastfield::FastValue;
+
+    use super::collector::PartitionValues;
+
+    #[inline(always)]
+    pub fn partitions_size_in_bytes<
+        TFastValue: FastValue + Display,
+        TPartitionFastValue: FastValue,
+    >(
+        partitions: &[PartitionValues<TFastValue, TPartitionFastValue>],
+    ) -> usize {
+        let mut size = 0;
+        for partition in partitions {
+            size += partition_size_in_bytes(partition);
+        }
+        size
+    }
+
+    #[inline(always)]
+    fn partition_size_in_bytes<TFastValue: FastValue + Display, TPartitionFastValue: FastValue>(
+        partition: &PartitionValues<TFastValue, TPartitionFastValue>,
+    ) -> usize {
+        std::mem::size_of::<TFastValue>() * partition.fast_field_values.len()
+            + std::mem::size_of::<u64>()
+            + std::mem::size_of::<TPartitionFastValue>()
+    }
+}
+
 #[cfg(test)]
```

Comment:
Would be nice to add a small test on the serialization part. One like this would be perfect.

```rust

#[test]
fn test_serialize_partitions() {
    let mut buffer = Vec::new();
    let partition_1 = PartitionValues {
        partition_value: 1u64,
        fast_field_values: vec![3u64, 4u64],
    };
    let partition_2 = PartitionValues {
        partition_value: 2u64,
        fast_field_values: vec![5u64],
    };
    serialize_partitions::<u64, u64>(&vec![partition_1, partition_2], &mut buffer).unwrap();
    let expected_buffer = vec![
        1u64.to_le_bytes(),
        16usize.to_le_bytes(),
        3u64.to_le_bytes(),
        4u64.to_le_bytes(),
        2u64.to_le_bytes(),
        8usize.to_le_bytes(),
        5u64.to_le_bytes(),
    ]
    .into_iter()
    .flatten()
    .collect_vec();
    assert_eq!(buffer, expected_buffer);
}
```